### PR TITLE
MRG, FIX: Link to RandomState properly

### PR DESCRIPTION
**Pull request by larsoner**
Also a tiny fix for a short-circuit in `SphinxDocLinkResolver.resolve`.

### CODE REVIEW

**Comment by larsoner** [comment]
Okay I think this is finally ready @lucyleeow. The tests pass and the links look correct to me here:

https://1238-25860190-gh.circle-artifacts.com/0/tiny_html/auto_examples/plot_numpy_matplotlib.html

File: sphinx_gallery/docs_resolv.py [hunk 1]
```diff
@@ -362,9 +361,8 @@ def _embed_code_links(app, gallery_conf, gallery_dir):
                                    not cobj['is_class'])
```
Comment (lucyleeow) [inline sphinx_gallery/docs_resolv.py]: Might be worth putting a note here to say CSS classes are being added here?

Comment (lucyleeow) [inline sphinx_gallery/docs_resolv.py]: Might be worth putting a note here to say CSS classes are being added here?

File: sphinx_gallery/tests/tinybuild/examples/plot_numpy_matplotlib.py [hunk 2]
```diff
@@ -37,7 +37,7 @@
 warn('This warning should show up in the output', RuntimeWarning)
 x = Figure()  # plt.Figure should be decorated (class), x shouldn't (inst)
 # should not crash, nested resolution, but doesn't resolve because
```
Comment (lucyleeow) [inline sphinx_gallery/tests/tinybuild/examples/plot_numpy_matplotlib.py]: Amend the note as it does resolve now?

**Comment by lucyleeow** [comment]
Honestly it still confuses me after looking at it all day but seems fine as far as I can tell.

**Comment by larsoner** [comment]
Pushed a commit to add comments, okay to merge once CIs come back happy @lucyleeow ? Then I'll probably tackle the next item or two on #587

**Comment by lucyleeow** [comment]
To the extent of my reviewing abilities - yes! Merge away.

**Pull request merged**